### PR TITLE
HomeWork 1 100%

### PR DESCRIPTION
**Pull request by gowgow1**
**my App**
image
self-esteem: 
1. [1 point] - The architecture of the application should be the same as presented above. +
2. [0.5 point] - Components are presented as function components (not class components). +
3. [0.5 point] - Use .jsx extensions for files with jsx syntax. +
4. [1 point] - All inputs and buttons should be reusable components. +
5. [2 points] - Implement searching functionality by title and id. +
6. [1 point] - Show list of courses. +
7. [1 point] - CourseCard component should be rendered inside Courses component using a loop. +
8. [6 points] - Add logic for creating courses. 
[0.5 point] - Possibility to add a title. +
[0.5 point] - Possibility to add a description. +
[1 point] - Possibility to create a new authors. +
[1 point] - Possibility to add a duration. +
[1 point] - Show duration time in a format «mm.ss». +
[1 point] - Add logic for creating new author, adding author to Course authors and deleting author +
[1 point] - Add logic for saving course (new course should be presented in the courses list). +
9. [2 points] - eslint, prettier and pre-commit hook work correctly. -+ ((husky failed if instal it according advices of courses)

### CODE REVIEW

File: courses-app/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import Header from './components/Header';
+import Courses from './components/Courses';
+import CreateCourse from './components/CreateCourse';
+
+import { mockedAuthorsList, mockedCoursesList } from './constants';
+
+import './App.css';
+
+function App() {
+	const [showCoursePage, toggleCoursePage] = useState(false);
```
Comment (DrDiman) [inline courses-app/src/App.jsx:12]: talk about naming

File: courses-app/src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	const updateAuthors = (newAuthor) => {
+		setAuthorsList([...authorsList, newAuthor]);
+	};
+
+	const updateCourses = (newCourse) => {
+		setCourseList([...coursesList, newCourse]);
+		togglePage();
+	};
+
+	const togglePage = () => toggleCoursePage(!showCoursePage);
```
Comment (DrDiman) [inline courses-app/src/App.jsx:25]: callback as a setter

File: courses-app/src/components/Courses/components/CourseCard/CourseCard.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import Button from '../../../common/Button';
+import timeFormat from '../../../../helpers/timeFormat';
```
Comment (DrDiman) [inline courses-app/src/components/Courses/components/CourseCard/CourseCard.jsx:2]: talk about `@` aliases

File: courses-app/src/components/Courses/components/CourseCard/CourseCard.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+				{created}
+			</div>
+			<div className='show-btn-wrap'>
+				<Button text='Show course' />
+			</div>
+		</div>
+	</div>
+);
+
+export default CourseCards;
```
Comment (DrDiman) [inline courses-app/src/components/Courses/components/CourseCard/CourseCard.jsx:39]: nice to have named export import everywhere and PropTypes checks

File: courses-app/src/components/CreateCourse/index.css [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+.control {
+    display: flex;
+    justify-content: space-between;
+    margin-bottom: 15px;
+}
+
+
+
```
Comment (DrDiman) [inline courses-app/src/components/CreateCourse/index.css:23]: do not skip lines)

File: courses-app/src/components/CreateCourse/index.css [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+.duration-bar span {
+	font-size: 20px;
+	font-weight: 600;
+}
+
+h2 {
+		font-size: 20px;
+		text-align: center;
```
Comment (DrDiman) [inline courses-app/src/components/CreateCourse/index.css:55]: something with indents

File: courses-app/src/components/CreateCourse/index.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+import CreateCourse from './CreateCourse';
+export default CreateCourse;
```
Comment (DrDiman) [inline courses-app/src/components/CreateCourse/index.js:2]: could be as one line

`export * from './CreateCourse'`

File: .editorconfig [hunk 8]
```diff
@@ -0,0 +1,3 @@
+[*]
```
Comment (DrDiman) [inline .editorconfig:1]: about husky:

install as devDependencies `husky`

and add e.g. such key in package.json:
```
  "husky": {
    "hooks": {
      "pre-push": "yarn lint"
    }
  }
  ```

File: courses-app/src/components/CreateCourse/CreateCourse.jsx [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+				<Input
+					placeholder='Enter Title'
+					labelText='Title'
+					id='title'
+					onChange={handleInputName}
+					name='title'
+				/>
+				<Button
+					text='Create course'
+					onClick={(event) => {
```
Comment (DrDiman) [inline courses-app/src/components/CreateCourse/CreateCourse.jsx:47]: do not use inline handlers

File: courses-app/src/components/CreateCourse/CreateCourse.jsx [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+						placeholder='Enter author name...'
+						labelText='Author name:'
+						id='authorInput'
+						name='authorInput'
+						onChange={handleInputName}
+					/>
+					<Button text='Create Author' onClick={update} />
+					<h2>Duration</h2>
+					<Input
+						placeholder='Enter duration in minutes...'
```
Comment (DrDiman) [inline courses-app/src/components/CreateCourse/CreateCourse.jsx:81]: talk about localization

File: courses-app/src/App.jsx (hunk 2 above)
Comment (gowgow1) [reply courses-app/src/App.jsx:25]: removed;)

File: courses-app/src/components/Courses/components/CourseCard/CourseCard.jsx (hunk 3 above)
Comment (gowgow1) [reply courses-app/src/components/Courses/components/CourseCard/CourseCard.jsx:2]: it's a good advice for future, thank you.

File: courses-app/src/components/CreateCourse/index.css (hunk 6 above)
Comment (gowgow1) [reply courses-app/src/components/CreateCourse/index.css:55]: fixed

File: courses-app/src/components/CreateCourse/index.js (hunk 7 above)
Comment (gowgow1) [reply courses-app/src/components/CreateCourse/index.js:2]: fixed over all porject

File: .editorconfig (hunk 8 above)
Comment (gowgow1) [reply .editorconfig:1]: idk, but it doesn't work :(

**Pull request closed without merging**